### ISSUE TITLE
FutureWarning: FSDP.state_dict_type() and FSDP.set_state_dict_type() are being deprecated. Please use APIs, get_state_dict() and set_state_dict(), which can support different parallelisms, FSDP1, FSDP2, DDP.

### ISSUE THREAD

**Issue by NanoCode012** [axolotl-ai-cloud/axolotl#2402]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [x] I searched previous [Ideas in Discussions](https://github.com/axolotl-ai-cloud/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [x] I searched previous [Issues](https://github.com/axolotl-ai-cloud/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

```
FutureWarning: FSDP.state_dict_type() and FSDP.set_state_dict_type() are being deprecated. Please use APIs, get_state_dict() and set_state_dict(), which can support different parallelisms, FSDP1, FSDP2, DDP. API doc: https://pytorch.org/docs/stable/distributed.checkpoint.html#torch.distributed.checkpoint.state_dict.get_state_dict .Tutorial: https://pytorch.org/tutorials/recipes/distributed_checkpoint_recipe.html .
```

### ✔️ Solution

https://pytorch.org/tutorials/recipes/distributed_checkpoint_recipe.html 
https://pytorch.org/docs/stable/distributed.checkpoint.html#torch.distributed.checkpoint.state_dict.get_state_dict

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this feature has not been requested yet.
- [x] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by NanoCode012** [comment]
Forgot where this was thrown, but likely here https://github.com/axolotl-ai-cloud/axolotl/blob/80304c26a70e21ed8522fdbd53bcb290f9c6b7d3/src/axolotl/train.py#L246

**Comment by ved1beta** [comment]
[consider closing ](https://github.com/axolotl-ai-cloud/axolotl/pull/3083#discussion_r2288453806)

**Comment by NanoCode012** [comment]
Sure, thanks for the followup
